lsp/server/src/diagnostic/DiagnosticProducer.ts

```diff
@@ -19,4 +26,9 @@ export interface DiagnosticProducer<T> {
         textDocument: TextDocument,
         astNode: T
     ): Promise<Diagnostic[]>;
+
+}
+
+export interface DiagnosticMetaData {
+    id: DiagnosticId
```

Comment:
the id is used to implement the Suppress this diagnostic CodeAction in quickfix.